samples/crud_extension/crud_handler.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+#
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+from opensearchpy import OpenSearch
+from opensearch_sdk_py.rest.extension_rest_handler import ExtensionRestHandler
+from opensearch_sdk_py.rest.named_route import NamedRoute
+from opensearch_sdk_py.rest.rest_method import RestMethod
+from opensearch_sdk_py.rest.rest_status import RestStatus
+from opensearch_sdk_py.rest.extension_rest_response import ExtensionRestResponse
+from opensearch_sdk_py.rest.extension_rest_request import ExtensionRestRequest
+
+import logging
+import json
+
+logging.basicConfig(encoding="utf-8", level=logging.INFO)
+
+class CRUDRestHandler(ExtensionRestHandler):
+    def __init__(self) -> None:
+        self.client = OpenSearch(
+            hosts=[{'host': 'localhost', 'port': 9200}],
+            http_auth=('admin', 'admin'),  # For testing only. Don't store credentials in code.
+            use_ssl=False,
+            verify_certs=False,
+            ssl_show_warn=False
+        )
+
+    def handle_request(self, request: ExtensionRestRequest) -> ExtensionRestResponse:
+        logging.info(f"handling {request}")
+        logging.info(f"request method: {request.method}")
+
+        if request.method == RestMethod.POST:
+            # Create a document
+            index_name = 'test-index'
+            document = request.content
+            response = self.client.index(
+                index=index_name,
+                body=document,
+                refresh=True
+            ) 
+            response_bytes = bytes(json.dumps(response).encode("utf-8"))
+            return ExtensionRestResponse(RestStatus.OK, response_bytes, ExtensionRestResponse.JSON_CONTENT_TYPE)
+        
+        elif request.method == RestMethod.GET:
+            # Search for documents
+            index_name = 'test-index'
+            q = request.params.get('q')
+            
+            query = {
+                'size': 5,
+                'query': {
+                    'multi_match': {
+                        'query': q,
+                        'fields': ['title^2', 'director']
+                    }
+                }
+            }
+            response = self.client.search(
+                body=query,
+                index=index_name
+            )
+            response_bytes = bytes(json.dumps(response).encode("utf-8"))
+            return ExtensionRestResponse(RestStatus.OK, response_bytes, ExtensionRestResponse.JSON_CONTENT_TYPE, consumed_params=['q'])
```

Comment:
(Style preference) Would prefer you create a consumed_params list and then append to it at the time you access the param.  Not a big deal as it'll all get fixed by #49 eventually but if you're making other changes it would make it more readable here.  Same comment for `id` in other classes.